### ISSUE TITLE
Multiple bots 

### ISSUE THREAD

**Issue by edos4** [jgorset/facebook-messenger#262]
Can you please provide an example for a bot for 2 facebook pages?

On the README,
1. Where do we put the ExampleProvider class?
2. Where do we put the multiple ACCESS_TOKEN, APP_SECRET, and VERIFY_TOKEN for each pages? 
3. How does the bot differentiate which page the message originated?

**Comment by edos4** [comment]
Starting to get the hang of it. Your gem is amazing, @jgorset !

On #3, 
I have 2 pages. For example, the page IDs from developers.facebook.com are: 111111111111111 and 222222222222222. When the bot receives a message from `Bot.on :message do |message|`, I could get the page ID where it originated by `message.recipient['id']` then reply accordingly. 

Am I doing this right so far?

**Comment by jgorset** [comment]
Hi @edos4, and thank you! Glad it's helpful to you!

I guess you've already figured out that you need to implement your own configuration provider that doesn't use those default environment variables (`ACCESS_TOKEN`, `APP_SECRET` and `VERIFY_TOKEN`), but some other method that allows you to authenticate with different credentials for different pages. I've written a little about it under ["Make a configuration provider" in the README](https://github.com/jgorset/facebook-messenger#make-a-configuration-provider). I hope that put you on the right track, and otherwise please make a pull request that improves upon it for the next person.

>I have 2 pages. For example, the page IDs from developers.facebook.com are: 111111111111111 and 222222222222222. When the bot receives a message from Bot.on :message do |message|, I could get the page ID where it originated by message.recipient['id'] then reply accordingly.

That's exactly right, but if you're replying immediately (with `message.reply` inside a `Bot.on` block), the gem will already figure all of that out for you! If you look at [the code for `message.reply`](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/incoming/common.rb#L119), you'll find that it gets around to calling [`access_token`](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/incoming/common.rb#L134) and [`app_secret_proof`](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/incoming/common.rb#L138), which in turn calls the `app_secret_for(id)` and `app_secret_proof_for(id)` methods of -- you guessed it -- your configuration provider!

If you are sending messages that aren't a direct response to a message that you receive (with `Bot.deliver`), though, you will need to specify `access_token` and `app_secret_proof` as arguments, because the gem needs to know which page you want to send the message from.

**Comment by jgorset** [comment]
I got around to #259 now, so if you want to use just `page_id` instead of messing around with access tokens outside of your configuration provider, now you can!

It's a backwards incompatible change, so I released it with version 2.0.0.

------

The 1.x way:

```ruby
Bot.deliver({
  recipient: {
    id: '45123'
  },
  message: {
    text: 'Human?'
  },
  messaging_type: Facebook::Messenger::Bot::MessagingType::UPDATE
}, access_token: ENV['ACCESS_TOKEN'], app_secret_proof: app_secret_proof)
```

The 2.x way:
```ruby
Bot.deliver({
  recipient: {
    id: '45123'
  },
  message: {
    text: 'Human?'
  },
  messaging_type: Facebook::Messenger::Bot::MessagingType::UPDATE
}, page_id: YOUR_PAGE_ID)
```
